plugins/grpc-transport-auth/src/test/java/com/github/bsideup/liiklus/transport/grpc/config/GRPCAuthConfigTest.java

```diff
@@ -0,0 +1,93 @@
+package com.github.bsideup.liiklus.transport.grpc.config;
+
+import com.github.bsideup.liiklus.transport.grpc.GRPCLiiklusTransportConfigurer;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.context.properties.bind.validation.BindValidationException;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.support.StaticApplicationContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class GRPCAuthConfigTest {
+    private static final String FULL_2048 = "-----BEGIN PUBLIC KEY-----\n" +
+            "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6b/6nQLIQQ8fHT4PcSyb\n" +
+            "hOLUE/237dgicbjsE7/Z/uPffuc36NTMJ122ppz6dWYnCrQ6CeTgAde4hlLE7Kvv\n" +
+            "aFiUbe5XKwSL8KV292XqrwRZhMI58TTTygcrBodYGzHy0Yytv703rz+9Qt5HO5BF\n" +
+            "02/+sM+Z0wlH6aXl3K3/2HfSOfitqnArBGaAs+PRNX2jlVKD1c9Cb7vo5L0X7q+6\n" +
+            "55uBErEoN7IHbj1u33qI/xEvPSycIiT2RXMGZkvDZH6mTsALel4aP4Qpp1NcE+kD\n" +
+            "itoBYAPTGgR4gBQveXZmD10yUVgJl2icINY3FvT9oJB6wgCY9+iTvufPppT1RPFH\n" +
+            "dQIDAQAB\n" +
+            "-----END PUBLIC KEY-----\n";
+
+    ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner(() -> new StaticApplicationContext() {
+        @Override
+        public void refresh() throws BeansException, IllegalStateException {
+        }
+    })
+            .withInitializer((ApplicationContextInitializer) new GRPCAuthConfig());
+
+    @Test
+    void shouldRequireAlg() {
+        applicationContextRunner = applicationContextRunner.withPropertyValues(
+                "spring.profiles.active: not_gateway",
+                "grpc.auth.alg: NONE"
+        );
+        applicationContextRunner.run(context -> {
+            assertThat(context).doesNotHaveBean(GRPCLiiklusTransportConfigurer.class);
+        });
+    }
+
+    @Test
+    void shouldAddWithHmac512() {
+        applicationContextRunner = applicationContextRunner.withPropertyValues(
+                "spring.profiles.active: not_gateway",
```

Comment:
`not_gateway` but has a bean? Huh?

Reply:
should I require gateway profile? I would expect that it wouldn't be a situation when grpc.auth has some setup, but grpc itself is disabled. That's why I simplified this config and left it independently from the anything else.